RepoMan/RepoMan/Repository/GitHubPullRequestReaderFactory.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using Octokit;
+using RepoMan.Analysis.Normalization;
+using RepoMan.Records;
+
+namespace RepoMan.Repository
+{
+    public class GitHubPullRequestReaderFactory :
+        IPullRequestReaderFactory
+    {
+        private readonly string _productHeaderValue;
+        private readonly INormalizer _bodyNormalizer;
+        private readonly Dictionary<string, GitHubClient> _clientsByApiKey;
+        
+        public GitHubPullRequestReaderFactory(string productHeaderValue, INormalizer bodyNormalizer)
+        {
+            if (string.IsNullOrWhiteSpace(productHeaderValue)) throw new ArgumentNullException(nameof(productHeaderValue));
+            _productHeaderValue = productHeaderValue;
+            _bodyNormalizer = bodyNormalizer ?? throw new ArgumentNullException(nameof(bodyNormalizer));
+            _clientsByApiKey = new Dictionary<string, GitHubClient>(StringComparer.Ordinal);
+        }
+
+        private GitHubClient GetClient(WatchedRepository repo)
+        {
+            if (string.IsNullOrWhiteSpace(repo.ApiToken)) throw new ArgumentNullException(nameof(repo.ApiToken));
+            if (string.IsNullOrWhiteSpace(repo.Url)) throw new ArgumentNullException(nameof(repo.Url));
+
+            if (_clientsByApiKey.TryGetValue(repo.ApiToken, out var existingClient))
+            {
+                return existingClient;
+            }
```

Comment:
@bojanrajkovic - this is where/how clients are reused. And over in the BBCloud implementation, too.